samples/client/petstore/javascript-flowtyped/lib/api.js

```diff
@@ -100,7 +100,7 @@ export var PetApiFetchParamCreator = function PetApiFetchParamCreator(configurat
       delete localVarUrlObj.search;
       localVarRequestOptions.headers = Object.assign({}, localVarHeaderParameter, options.headers);
       var needsSerialization = typeof body !== "string" || localVarRequestOptions.headers['Content-Type'] === 'application/json';
-      localVarRequestOptions.body = needsSerialization ? JSON.stringify(body != null ? body : {}) : body || "";
+      localVarRequestOptions.body = needsSerialization ? JSON.stringify(body) : body;
```

Comment:
@DanielRuf seems like the `{}` case will be omitted when body is null, right?

Reply:
`body` is already checked earlier (against null).

Reply:
See https://github.com/DanielRuf/openapi-generator/blob/1b2a25f331ad53e632c287f76aacf256009f3626/samples/client/petstore/javascript-flowtyped/lib/api.js#L79

Reply:
What if the body parameter is optional instead of required?

Reply:
It isn't currently according to the code at https://github.com/DanielRuf/openapi-generator/blob/1b2a25f331ad53e632c287f76aacf256009f3626/samples/client/petstore/javascript-flowtyped/lib/api.js#L79

SonarQube and other tools and linters confirm this.

Reply:
Petstore spec doesn't cover all use cases. Please also consider the use case in which the body parameter is optional.

Reply:
As already described, checking it twice makes not much sense. Please see the line 39 and so on. Most linters also report this (SonarCsnner for example).

Reply:
I never said your fix doesn't make sense. We simply need to consider different cases, e.g. required vs optional body parameter.

You may find the `{{#required}} ... {{/required}}` mustache tag useful.

Reply:
Well, but this would be a different PR then and also some refactoring. Nothing that I would directly do in this cleanup / chore / fix PR. I would separate them.

Reply:
Sorry we cannot accept a partial fix that breaks the optional body parameters.

If you want to continue working on this PR, please feel free to reopen it with a fix that works for both required and optional body parameters.

Reply:
> Sorry we cannot accept a partial fix that breaks the optional body parameters.

Not sure how it should break here (as described in the other comments). I checked the code and this would not break. And as already stated a `git commit --amend` would be possible from my side ;-)

The changes were only in parts where it was already checked.
Let me clarify this:

https://github.com/DanielRuf/openapi-generator/blob/1b2a25f331ad53e632c287f76aacf256009f3626/samples/client/petstore/javascript-flowtyped/lib/api.js#L78-L81

https://github.com/DanielRuf/openapi-generator/blob/1b2a25f331ad53e632c287f76aacf256009f3626/samples/client/petstore/javascript-flowtyped/lib/api.js#L267-L270

https://github.com/DanielRuf/openapi-generator/blob/1b2a25f331ad53e632c287f76aacf256009f3626/samples/client/petstore/javascript-flowtyped/lib/api.js#L634-L637

https://github.com/DanielRuf/openapi-generator/blob/1b2a25f331ad53e632c287f76aacf256009f3626/samples/client/petstore/javascript-flowtyped/lib/api.js#L751-L754

https://github.com/DanielRuf/openapi-generator/blob/1b2a25f331ad53e632c287f76aacf256009f3626/samples/client/petstore/javascript-flowtyped/lib/api.js#L782-L785

https://github.com/DanielRuf/openapi-generator/blob/1b2a25f331ad53e632c287f76aacf256009f3626/samples/client/petstore/javascript-flowtyped/lib/api.js#L813-L816

https://github.com/DanielRuf/openapi-generator/blob/1b2a25f331ad53e632c287f76aacf256009f3626/samples/client/petstore/javascript-flowtyped/lib/api.js#L971-L974

Same for the other file as far as I can see.

https://github.com/DanielRuf/openapi-generator/blob/1b2a25f331ad53e632c287f76aacf256009f3626/samples/client/petstore/javascript-flowtyped/src/api.js#L300-303

https://github.com/DanielRuf/openapi-generator/blob/1b2a25f331ad53e632c287f76aacf256009f3626/samples/client/petstore/javascript-flowtyped/src/api.js#L492-L495

https://github.com/DanielRuf/openapi-generator/blob/1b2a25f331ad53e632c287f76aacf256009f3626/samples/client/petstore/javascript-flowtyped/src/api.js#L864-L867

https://github.com/DanielRuf/openapi-generator/blob/1b2a25f331ad53e632c287f76aacf256009f3626/samples/client/petstore/javascript-flowtyped/src/api.js#L983-L986

https://github.com/DanielRuf/openapi-generator/blob/1b2a25f331ad53e632c287f76aacf256009f3626/samples/client/petstore/javascript-flowtyped/src/api.js#L1013-L1016

https://github.com/DanielRuf/openapi-generator/blob/1b2a25f331ad53e632c287f76aacf256009f3626/samples/client/petstore/javascript-flowtyped/src/api.js#L1043-L1046

https://github.com/DanielRuf/openapi-generator/blob/1b2a25f331ad53e632c287f76aacf256009f3626/samples/client/petstore/javascript-flowtyped/src/api.js#L1187-L1190

So I see no optional `body` there.

> Sorry we cannot accept a partial fix that breaks the optional body parameters.

Not sure what you mean with 
> Sorry we cannot accept a partial fix that breaks the optional body parameters.